### DISCUSSION TITLE
How to pass values as out from C# -> JS interop?

### DISCUSSION THREAD

**Discussion by nathan130200** [sebastienros/jint#1178] [Q&A] [answered]
I'm creating an external script that should access `Dictionary<K, V>` but the point is. `bool TryGetValue(K key, out V value)` theres no way to pass variables as reference in JS.

Some ideas how to deal with that?
Or i should create something like an proxyied object?

**Comment by lahma** [comment, marked as answer]
`out` and `ref` share the the same problem, JS doesn't have such semantics that would map well. This get awfully complex with all the closures etc. So unfortunately you need to add helpers, like extension method `GetValueOrDefault` (which I think is also part of latest NETversions), Jint does support registering extension methods too!

**Comment by nathan130200** [comment]
Oh. Thanks for answer.
